x-pack/plugins/cloud/common/register_cloud_deployment_id_analytics_context.ts

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { AnalyticsClient } from '@kbn/analytics-client';
+import { of } from 'rxjs';
+
+export function registerCloudDeploymentIdAnalyticsContext(
+  analytics: Pick<AnalyticsClient, 'registerContextProvider'>,
+  cloudId?: string
```

Comment:
Noob question: In what case(s) is this value `undefined`? Does it make sense to register context without any data?

Reply:
> In what case(s) is this value `undefined`?

The `cloud` plugin is always enabled. However, it has a `cloudId` only if actually deployed on ESS.

> registering it will cause the `analytics_client` to create a subscription and a whole bunch of stuff pretty much for nothing. Perhaps we could check if `cloudId` is provided in this very same method (sorry if I am missing something here), WDYT?

It's a good performance improvement indeed. Applied the changes :)